### PR TITLE
[ResNet] Upgrade to Burn 0.13.0

### LINKED ISSUE #19: [ResNet] Chore: Use Candle's published crate version instead of git revision
Currently, the `Cargo.toml` points to a git revision of Burn required to properly import PyTorch weights.

Once `candle-core` includes the patches in a release version and Burn includes that version in a release, we should update the versions.

```
[dependencies]
# Note: default-features = false is needed to disable std
burn = { version = "0.13.0", default-features = false }
burn-import = "0.13.0"
dirs = { version = "5.0.1", optional = true }
serde = { version = "1.0.192", default-features = false, features = [
    "derive",
    "alloc",
] } # alloc is for no_std, derive is needed

[dev-dependencies]
burn = {  version = "0.13.0", features = ["ndarray"] }
image = { version = "0.24.7", features = ["png", "jpeg"] }
```

### PR DESCRIPTION
**Pull request by laggui**
Closes #19 

- Updated `burn` and `image` versions
- Changed `ReLU` -> `Relu`
- Removed `init_with` methods (yay lazy init)
- Refactored residual blocks to use newly supported enum modules
- Changes `burn-rs` -> `tracel-ai` github links
- Added `with_classes` method to support output layer initialization for fine-tuning

### CODE REVIEW

**Comment by laggui** [comment]
Forgot I don't need a review approval on the models repo to merge.. will still wait for at least one review to merge for sanity :)

**Review by antimora — APPROVED** [review]
LGTM

**Pull request merged**
